src/ray/raylet/worker_pool.cc

```diff
@@ -115,11 +116,28 @@ void WorkerPool::StartWorkerProcess(const Language &language) {
                  << state.idle_actor.size() << " actor workers, and " << state.idle.size()
                  << " non-actor workers";
 
+  std::string prefix = "";
+  std::string suffix = "";
+  if (task_spec != nullptr && task_spec->IsActorCreationTask()) {
+    prefix = task_spec->WorkerStartingPrefix();
+    suffix = task_spec->WorkerStartingSuffix();
+  }
+
   // Extract pointers from the worker command to pass into execvp.
   std::vector<const char *> worker_command_args;
+  if (!prefix.empty()) {
+    worker_command_args.push_back(prefix.c_str());
+  }
   for (auto const &token : state.worker_command) {
     worker_command_args.push_back(token.c_str());
   }
+  // Note that this is used for Java worker only currently.
+  if (!suffix.empty() && task_spec->GetLanguage() == Language::JAVA) {
+    RAY_CHECK(worker_command_args.size() > 2)
+        << "At least 2 elements of the Java worker command.";
+    const auto pos_to_insert = worker_command_args.size() - 1;
+    worker_command_args.insert(worker_command_args.begin() + pos_to_insert, suffix.c_str());
+  }
```

Comment:
Is this because JVM option have to come before the last argument (the main class)?
If so, it's better to put some a placeholder in the worker command?